server/src/test/java/org/elasticsearch/transport/RemoteConnectionManagerTests.java

```diff
@@ -89,6 +89,19 @@ public void testGetConnection() {
         assertEquals(1, versions.size());
     }
 
+    public void testResolveRemoteClusterAlias() {
```

Comment:
Would also cover the scenario where the connection needs to be unwrapped -- either randomized or as a third case below.

Reply:
Good point. I'll add a new case for proxied connections.

Reply:
Added test in https://github.com/elastic/elasticsearch/pull/91724/commits/c4043826a12798080601076b98bf16b44e1231d3